rfcs/2022-02-24-11552-dd-agent-style-secret-management.md

```diff
@@ -0,0 +1,191 @@
+# RFC 11552 - 2022-02-24 - Datadog Agent style secret management
+
+The Datadog Agent has a straighforward secret resolution facility to avoid having sensitive information stored directly
+in its config, it relies on a user-provided external program that is run to retrieve sensitive value from a third party
+system. This RFC aims to propose a similar mechanism for Vector.
+
+## Context
+
+The [Datadog Agent documentation][dd-agent-secret-mgmt] provide all user-relevant information to use that feature. It
+covers the specification for the user-provided executable that loads/decrypts secrets along with the agent configuration
+and the syntax to retrieve encrypted config value.
+
+## Cross cutting concerns
+
+- The ongoing [configuration schema work][vector-config-schema-work].
+- `vector config` and related Vector enterprise work.
```

Comment:
```suggestion
- `vector config` ([PR][vector-config-subcommand])and related Vector enterprise work.
```

(In conjunction with a link to be added to the doc, referencing https://github.com/vectordotdev/vector/pull/11442)